### DISCUSSION TITLE
How do I get domain name during generate sitemap

### DISCUSSION THREAD

**Discussion by naseef0** [vercel/next.js#50747] [App Router]
```
export default function sitemap() {
  return [
    {
      url: 'https://acme.com',
      lastModified: new Date(),
    },
    {
      url: 'https://acme.com/about',
      lastModified: new Date(),
    },
    {
      url: 'https://acme.com/blog',
      lastModified: new Date(),
    },
  ];
}

```

From the above example, I need  `url`  as dynamic. We can set url as an environment variable. But if we consider vercel deployment, vercel will provide different domain names. It won't work, right?

![image](https://github.com/vercel/next.js/assets/45784156/353f80f5-7f36-4d75-8890-9830e503731c)

Please let me know if any workaround

**Comment by devjiwonchoi** [comment]
Assuming you've already set up accessing the `URL` params, Vercel deployment provides two types of `hostname` as follows:
- Deployment
- Production

For example, the `deployment hostname` will be [assigned per deployment](https://vercel.com/docs/concepts/deployments/generated-urls), as you have stated.

> Whenever a new [deployment](https://vercel.com/docs/concepts/deployments/overview) is created, Vercel will automatically generate a unique URL that is available publicly, or [password protected](https://vercel.com/docs/concepts/deployments/deployment-protection#password-protection).

On the other hand, the `production hostname` is [static by default](https://vercel.com/docs/concepts/projects/domains/working-with-domains) and it will not change unless you modify it.

> When you create a deployment on Vercel, we automatically assign it a domain based on your project name and ending in .vercel.app. Your site will be available to anyone that you share the domain with.
